lxd/storage/drivers/volume.go

```diff
@@ -173,3 +173,8 @@ func (v Volume) Snapshots(op *operations.Operation) ([]Volume, error) {
 
 	return snapVols, nil
 }
+
+// IsBlockBacked indicates whether storage device is block backed.
+func (v Volume) IsBlockBacked() bool {
+	return v.driver.Info().BlockBacking
```

Comment:
I think this should be BlockBacking && ContenType is filesystem.

Because if we're dealing with a block content type on a blockbacking storage driver (say a VM on LVM), we don't want to end up allowing the `block.*` config keys.

Reply:
@stgraber I did add that logic inside the common fields validation generator rather than here, because I didn't want to conflate the two.

Reply:
Indeed, was just about to comment here saying that this change isn't needed since you handle it properly in the common generator instead.